### ISSUE TITLE
python3.dll unnecessary with standalone_dynamic windows distribution

### ISSUE THREAD

**Issue by wkschwartz** [indygreg/PyOxidizer#359]
On Windows with MSVC, Create a minimal standalone-dynamic executable:

```bzl
def make():
    dist = default_python_distribution(flavor="standalone_dynamic") 
    files = FileManifest()
    files.add_python_resource(
        ".",
        dist.to_python_executable(
            name="pyapp",
            packaging_policy=dist.make_python_packaging_policy(),
            config=dist.make_python_interpreter_config(),
    ))
    return files

register_target("install", make, default=True)
resolve_targets()
```
After runnning `pyoxidizer build`, `\build\x86_64-pc-windows-msvc\debug\install` now contains `lib`, `pyapp.exe`, `python38.dll`, and `python3.dll`. Now we can use [`dumpbin`](https://docs.microsoft.com/en-us/cpp/build/reference/dumpbin-reference)  with [`/imports`](https://docs.microsoft.com/en-us/cpp/build/reference/imports-dumpbin)`:python3.dll` to verify that `python3.exe` is not actually used. Change into that directory and run

```console
> dumpbin pyapp.exe /imports:python3.dll
Microsoft (R) COFF/PE Dumper Version 14.26.28806.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file pyapp.exe

File Type: EXECUTABLE IMAGE

  Section contains the following imports:

  Summary

        3000 .data
       47000 .pdata
     1ABD000 .rdata
        2000 .reloc
        1000 .rsrc
      344000 .text
```
(If `python3.dll` were referenced, there would be a list of symbols under `Section contains the following imports:`. Verify this by using `/imports:python38.dll` above.) Repeat the above experiment replacing `pyapp.exe` with `python38.dll`, `lib\unicodedata.pyd`, `lib\_sqlite3.pyd`, and all the other `pyd`s and `dll`s under `lib`. None of them load `python3.dll`.

This is related to #353 insofar as PyOxidizer should be placing `vcruntime140.dll` rather than `python3.dll` next to the executable. (`python3.dll`'s presence is problematic for two reasons. The first is that it's clutter, thus defeating part of the purpose for compiling a Python app with PyOxidizer. The second is that it raises the cost of debugging if you know you have DLL problems but don't know that `python3.dll` isn't the source of your woe.)

**Comment by wkschwartz** [comment]
Immediately after posting this, I [found out](https://github.com/indygreg/PyOxidizer/issues/353#issuecomment-763157392) about the [`/dependents`](https://docs.microsoft.com/en-us/cpp/build/reference/dependents) option to `dumpbin`.

```console
> dumpbin pyapp.exe /dependents
Microsoft (R) COFF/PE Dumper Version 14.26.28806.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file pyapp.exe

File Type: EXECUTABLE IMAGE

  Image has the following dependencies:

    ADVAPI32.dll
    KERNEL32.dll
    python38.dll
    VCRUNTIME140.dll
    api-ms-win-crt-string-l1-1-0.dll
    api-ms-win-crt-heap-l1-1-0.dll
    api-ms-win-crt-utility-l1-1-0.dll
    api-ms-win-crt-convert-l1-1-0.dll
    api-ms-win-crt-runtime-l1-1-0.dll
    api-ms-win-crt-math-l1-1-0.dll
    api-ms-win-crt-stdio-l1-1-0.dll
    api-ms-win-crt-locale-l1-1-0.dll

  Summary

        3000 .data
       47000 .pdata
     1ABD000 .rdata
        2000 .reloc
        1000 .rsrc
      344000 .text
```

**Comment by wkschwartz** [comment]
The [PEP 425 ABI tag](https://www.python.org/dev/peps/pep-0425/#abi-tag) on wheels and extension module `.so`/`.dyld`/`.pyd` file names should be enough to determine whether `python3.dll` is required: If any wheel or extension module has the `abi3` tag, then it's needed. Otherwise it is not, if I understand correctly.

**Comment by indygreg** [comment]
Yes, we now always install `python3.dll` next to `pythonXY.dll`. This was the easiest to implemented. But you are correct that `python3.dll` is optional in many cases.

I agree we should only conditionally include `python3.dll`. However, this feels like a minor enhancement to me, so the current behavior may linger for a while.

Also, we'll want a Starlark config option to force its install because there will be applications that want to install/use extra Python packages as part of their normal running and they'll want to install `python3.dll` for maximum compatibility with future Python package installs (think use cases of software that supports extensions/plugins where the set of loaded code is not known at application build time).
